### ISSUE TITLE
Is it possible to add i/o disk metrics?

### ISSUE THREAD

**Issue by bhagenbourger** [cjbassi/gotop#19]
Please give the following info when creating an issue:

* Operating System
* CPU architecture (can be found with uname -m)
* Terminal
* Using tmux?
* Installation method

**Comment by cjbassi** [comment]
I changed the Disk Usage widget and added i/o to the latest release. It's a little hidden so let me know if it works for you or if it should be tweaked.

**Comment by bhagenbourger** [comment]
@cjbassi thanks, I test it on my own computer, I see R/s and W/s metrics well but unfortunately they are always set to 0, even if I read or write files from disk.

**Comment by cjbassi** [comment]
Hmm that's strange cause it's working for me. It might be an issue with gopsutil but it would be good to get some info first.

1. Is there more than one disk/partition listed in the Disk Usage widget?
2. Can you make sure that the partition you're checking in gotop is the one you're reading and writing to?
3. Is the other disk info correct?
4. What distro are you on?

Maybe you could try building from source to see if that does something too.

**Comment by bhagenbourger** [comment]
1. No I have only on disk listed
2. Yes, the percentage of occupation disk increases when I write data
3. Yes
4. Ubuntu 16.04 LTS

I'm not a Go developer so maybe I do something wrong to test. The steps I did are:
1. git clone into `~/go/src/github.com/cjbassi/gotop`
2. Switch to `diskio` branch
3. Execute `go run main.go`

How can I build from source?

**Comment by cjbassi** [comment]
You should test the `master` branch or download the latest binary using the `download.sh` script. The code in `diskio` is outdated and I'm gonna delete that branch actually. Let me know if anything changes!

Also, doing `go run main.go` is building/running from source :)

**Comment by bhagenbourger** [comment]
Yeah, I test from `master` branch and also using `download.sh`, both work! Thanks a lot, it's very useful :-)

**Comment by changchichung** [comment]
is this disk io metric enable by default ?
I install by download.sh , and there`s no disk io widget , or is there some other way to enable ?

**Comment by cjbassi** [comment]
It should be enabled by default. But the stats can sometimes be hidden if the window isn't wide enough.

**Comment by changchichung** [comment]
how wide ?
I see no diso io widget on screen in 1920x1080 fullscreen
or is there any way to check if the widget enabled ?

[screenshot](http://i.imgur.com/JAak1OE.png)

**Comment by changchichung** [comment]
OK , I see that , but the value always 0% bcz it show only loopX disks create by ubuntu snap
cant I select to show which disk?

**Comment by cjbassi** [comment]
Yah that would be a good feature request. Do you have a lot of those disks showing up?

**Comment by Lin-Buo-Ren** [comment]
@cjbassi Ubuntu-like system using a software distribution solution call [snaps](https://snapcraft.io), which is essentially a bunch of squashfs images that will be mounted on boot and thus spamming every block device monitor applications with the loop devices.

You should consider providing an option to exclude certain kind of block devices from showing, for example for `lsblk(1)`:

```
$ lsblk
NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
loop0    7:0    0   9.3M  1 loop /snap/snappy-debug/274
loop1    7:1    0  55.1M  1 loop /snap/snapcraft/2496
loop2    7:2    0  53.7M  1 loop /snap/core18/719
loop3    7:3    0    91M  1 loop /snap/core/6405
loop4    7:4    0  81.9M  1 loop /snap/youtube-dl/64
loop5    7:5    0   8.6M  1 loop /snap/asciinema/10
loop6    7:6    0    64K  1 loop /snap/hello/20
loop7    7:7    0  84.5M  1 loop /snap/youtube-dl/711
sda      8:0    0 447.1G  0 disk 
├─sda1   8:1    0  1007K  0 part 
├─sda2   8:2    0   190M  0 part /boot/efi
├─sda3   8:3    0  93.1G  0 part /media/Windows
├─sda4   8:4    0 353.8G  0 part /
└─sda5   8:5    0 839.5K  0 part /media/Security
sr0     11:0    1  1024M  0 rom  

$ lsblk --exclude 7
NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda      8:0    0 447.1G  0 disk 
├─sda1   8:1    0  1007K  0 part 
├─sda2   8:2    0   190M  0 part /boot/efi
├─sda3   8:3    0  93.1G  0 part /media/Windows
├─sda4   8:4    0 353.8G  0 part /
└─sda5   8:5    0 839.5K  0 part /media/Security
sr0     11:0    1  1024M  0 rom  
```

BTW, this application is featured in the [Snapcraft Live 6th Feb 2019 - YouTube](https://www.youtube.com/watch?v=BEp_l2oUcD8) video.

**Comment by cjbassi** [comment]
@Lin-Buo-Ren Snap mount points should be ignored but if they're showing for you then feel free to open an issue. Currently we use this to filter them out:

https://github.com/cjbassi/gotop/blob/3a804eb666cbc314f310a9c0fc94a2eef479b46e/src/widgets/disk.go#L65-L70

Also thanks for linking the video! Very cool to see that.
